CommonCPack.cmake

```diff
@@ -1,225 +1,276 @@
+  set(CPACK_COMPONENT_DOC_DEPENDS lib)
+
+  set(CPACK_COMPONENT_EXAMPLES_DISPLAY_NAME "${NAME} Examples")
+  set(CPACK_COMPONENT_EXAMPLES_DESCRIPTION "${NAME} Example Source Code")
+  set(CPACK_COMPONENT_EXAMPLES_DEPENDS dev)
+
+  set(CPACK_COMPONENT_LIB_DISPLAY_NAME "${NAME} Libraries")
+  set(CPACK_COMPONENT_LIB_DESCRIPTION "${NAME} Runtime Libraries")
 
   set(CPACK_COMPONENT_UNSPECIFIED_DISPLAY_NAME "Unspecified")
   set(CPACK_COMPONENT_UNSPECIFIED_DESCRIPTION
     "Unspecified Component - set COMPONENT in CMake install() command")
+endmacro()
+
+function(get_old_package_names PACKAGE_NAME ABI_VERSION OUTPUT)
+  math(EXPR _num_old_packages "${ABI_VERSION} - 1")
+  set(_old_packages)
+  foreach(i RANGE ${_num_old_packages})
+    list(APPEND _old_packages "${PACKAGE_NAME}${i}")
+  endforeach()
+  list(APPEND _old_packages "${PACKAGE_NAME}")
+  set(${OUTPUT} ${_old_packages} PARENT_SCOPE)
+endfunction()
+
+function(get_debian_arch OUTPUT)
+  # Get debian architecture name, e.g. amd64. Reference:
+  # https://www.debian.org/doc/debian-policy/ch-customized-programs.html#s-arch-spec
+  execute_process(COMMAND dpkg --print-architecture OUTPUT_VARIABLE _deb_arch
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  set(${OUTPUT} ${_deb_arch} PARENT_SCOPE)
+endfunction()
+
+macro(list_to_string LIST)
+  # Convert a list to a comma-separated string without duplicates
+  if(${LIST})
+    list(REMOVE_DUPLICATES ${LIST})
+    string(REGEX REPLACE ";" ", " ${LIST} "${${LIST}}")
+  endif()
+endmacro()
 
-  set(CPACK_COMPONENT_LIB_DISPLAY_NAME "${CPACK_PROJECT_NAME} Libraries")
-  set(CPACK_COMPONENT_LIB_DESCRIPTION "${CPACK_PROJECT_NAME} Runtime Libraries")
-
-  set(CPACK_COMPONENT_DEV_DISPLAY_NAME
-    "${CPACK_PROJECT_NAME} Development Files")
-  set(CPACK_COMPONENT_DEV_DESCRIPTION
-    "Header and Library Files for ${CPACK_PROJECT_NAME} Development")
-  set(CPACK_COMPONENT_DEV_DEPENDS lib)
-
-  set(CPACK_COMPONENT_DOC_DISPLAY_NAME "${CPACK_PROJECT_NAME} Documentation")
-  set(CPACK_COMPONENT_DOC_DESCRIPTION "${CPACK_PROJECT_NAME} Documentation")
-  set(CPACK_COMPONENT_DOC_DEPENDS lib)
+# Setup all CPACK variables
 
-  set(CPACK_COMPONENT_APPS_DISPLAY_NAME "${CPACK_PROJECT_NAME} Applications")
-  set(CPACK_COMPONENT_APPS_DESCRIPTION "${CPACK_PROJECT_NAME} Applications")
-  set(CPACK_COMPONENT_APPS_DEPENDS lib)
+common_cpack_set_base_parameters()
 
-  set(CPACK_COMPONENT_EXAMPLES_DISPLAY_NAME "${CPACK_PROJECT_NAME} Examples")
-  set(CPACK_COMPONENT_EXAMPLES_DESCRIPTION
-    "${CPACK_PROJECT_NAME} Example Source Code")
-  set(CPACK_COMPONENT_EXAMPLES_DEPENDS dev)
-elseif(CPACK_COMPONENTS_ALL STREQUAL "none")
-  set(CPACK_COMPONENTS_ALL)
+if(NOT CPACK_COMPONENTS_ALL)
+  common_cpack_set_components_all(${PROJECT_NAME})
 endif()
 
-include(LSBInfo)
-
 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-  find_program(RPM_EXE rpmbuild)
-  find_program(DEB_EXE debuild)
-endif()
+  # Legacy?
+  if(EXISTS ${PROJECT_SOURCE_DIR}/CMake/${PROJECT_NAME}.in.spec)
```

Comment:
It still does, along with Collage and Lunchbox. Looking at the file it must be for packaging with native debian tools, not CPack, so here is not the most appropriate place to put the configure() call but I am not sure where else to put it. What do think? Should I move it somewhere else or just remove it?